warehouse/query-core/src/main/java/datawave/query/jexl/nodes/ExceededOrThresholdMarkerJexlNode.java

```diff
@@ -131,24 +105,113 @@ public static JexlNode getExceededOrThresholdSource(JexlNode node) {
      * Get the parameters for this marker node (see constructors)
      * 
      * @param source
-     * @return A map of parameters for FIELD_PROP, and FST_URI_PROP or VALUES_PROP. URI will be a URI object, and values will be a Set of String
-     * @throws URISyntaxException
+     * @return The params associated with this ExceededOrThresholdMarker
+     * @throws IOException
      */
-    public static Map<String,Object> getParameters(JexlNode source) throws URISyntaxException {
-        Map<String,Object> returnParameters = new HashMap<>();
+    public static ExceededOrParams getParameters(JexlNode source) throws IOException {
         Map<String,Object> parameters = JexlASTHelper.getAssignments(source);
         // turn the FST URI into a URI and the values into a set of values
-        for (Map.Entry<String,Object> entry : parameters.entrySet()) {
-            if (entry.getKey().equalsIgnoreCase(FST_URI_PROP)) {
-                returnParameters.put(FST_URI_PROP, new URI(String.valueOf(entry.getValue())));
-            } else if (entry.getKey().equalsIgnoreCase(VALUES_PROP)) {
-                returnParameters.put(VALUES_PROP,
-                                new HashSet<>(Arrays.asList(StringUtils.split(String.valueOf(entry.getValue()), StringUtils.ESCAPE_CHAR, StringUtils.COMMA))));
-            } else if (entry.getKey().equalsIgnoreCase(FIELD_PROP)) {
-                returnParameters.put(FIELD_PROP, String.valueOf(entry.getValue()));
+        Object paramsObj = parameters.get(EXCEEDED_OR_PARAMS);
+        if (paramsObj != null)
+            return objectMapper.readValue(String.valueOf(paramsObj), ExceededOrParams.class);
+        else
+            return null;
+    }
+    
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public static class ExceededOrParams {
+        private String field;
+        private String fstURI;
+        private Collection<String> values;
+        private Collection<String[]> ranges;
+        
+        private ExceededOrParams() {
+            
+        }
+        
+        public ExceededOrParams(String field, String fstURI) {
+            this.field = field;
+            this.fstURI = fstURI;
+        }
+        
+        public ExceededOrParams(String field, Collection<String> values) {
+            this(field, values, null);
+        }
+        
+        public ExceededOrParams(String field, Collection<String> values, Collection<Range> ranges) {
+            this.field = field;
+            init(values, ranges);
+        }
+        
+        private void init(Collection<String> values, Collection<Range> ranges) {
+            if (ranges == null || ranges.isEmpty()) {
+                this.values = values;
+            } else {
+                SortedSet<Range> rangeSet = new TreeSet<>();
+                
+                if (values != null)
+                    values.forEach(value -> rangeSet.add(new Range(new Key(value), new Key(value))));
```

Comment:
`Range.exact(value)` is a more concise way to represent `new Range(new Key(value), new Key(value))` and also avoids the double instantiation.

Reply:
Nice!  I didn't know about that.  Updated in the latest commit.

Reply:
Actually, I need to revert this change.  Using Range.exact will add the null byte to the end key, which is something I don't want in this case.  Bummer!

Reply:
Oh good to know. My bad. Thanks!